internal/handlers/pg/pgdb/query.go

```diff
@@ -246,20 +246,38 @@ func prepareWhereClause(sqlFilters *types.Document) (string, []any) {
 	var p Placeholder
 
 	for k, v := range sqlFilters.Map() {
-		switch k {
-		case "_id":
-			switch v := v.(type) {
-			case string:
-				filters = append(filters, fmt.Sprintf(`((_jsonb->'_id')::jsonb = %s)`, p.Next()))
-				args = append(args, string(must.NotFail(pjson.MarshalSingleValue(v))))
+		// don't pushdown query operators and dot notation yet
+		if len(k) != 0 && (k[0] == '$' || types.NewPathFromString(k).Len() > 1) {
+			continue
+		}
 
-			case types.ObjectID:
+		// don't iterate through array for _id keys to simplify the query
+		if k == "_id" {
+			switch v := v.(type) {
+			case types.ObjectID, string:
```

Comment:
It looks to me that we only support string and ObjectID query pushdown for `_id` while we support more for other field? :thinking: Maybe I'm misunderstanding

Reply:
Exactly. As for `_id`s we support only strings and ObjectIDs, there's no need to send the SQL query that would check if `_id` field is an array.

This technically doesn't change much in performance, as PostgreSQL won't evaluate the whole query if it doesn't need to, although from logging perspecitve, and further debugging, it seems reasonable to keep queries as simple as possible

wdyt?